### PR TITLE
ci: add visual regressions bases on reg-viz

### PR DESCRIPTION
**Pull request by abykovssc**

### CODE REVIEW

**Comment by abykovssc** [comment]
For now it runs well locally, but need to run in docker on CI, otherwise screenshots have differences 🤦

**Comment by abykovssc** [comment]
And it's alive now! What's left is to add docker layer caching and proper artifact archival, and this is read for prime time

File: visual-regressions/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## What is this?
+
+This is a small package for running visual regressions in storybook locally
+
+This is separated from main docker image and package json,
+so some time can be saved building the docker image
+Normally you should never use this package directly, just use `ssc-ui-core` and launch
```
Comment (lhorak) [inline visual-regressions/README.md:7]: remove mention of ui-core

File: visual-regressions/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+`yarn test:storybook:visual`
+
+## Why ?
+To save time. We could use root Dockerfile from ssc-ui-core, but it installs all node_modules from ssc-ui-core,
```
Comment (lhorak) [inline visual-regressions/README.md:11]: ui-core

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,10 @@
 .DS_Store
 .idea/*
+.storybook/image-snapshots/actual
+.storybook/image-snapshots/diff
+.storybook/image-snapshots/report.html
+.storybook/image-snapshots/report.json
+public/storybook
```
Comment (ajkl2533) [inline .gitignore:7]: not sure about this path, storybook is built into the `storybook-static` folder

File: visual-regressions/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+## What is this?
```
Comment (ajkl2533) [inline visual-regressions/README.md:1]: Could you please move this into the https://github.com/securityscorecard/design-system/blob/main/TESTING.md#visual-regression

Comment (ajkl2533) [reply visual-regressions/README.md:1]: or just reference it there

File: package.json [hunk 5]
```diff
@@ -18,6 +18,9 @@
     "build": "rollup -c",
     "storybook": "start-storybook -p 8008",
     "storybook:build": "build-storybook --quiet",
+    "test:storybook:visual": "./visual-regressions/bin/take-and-check-in-docker.sh",
+    "test:storybook:visual:update": "./visual-regressions/bin/take-and-update-in-docker.sh",
+    "test:storybook:visual:ci": "build-storybook -o 'public/storybook' && ./visual-regressions/bin/take-and-check-in-docker-ci.sh",
```
Comment (ajkl2533) [inline package.json:23]: could you make it `build-storybook --quiet ...` it's making a mess in CI logs

File: visual-regressions/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,7 @@
+FROM abykovssc/visual-regressions
```
Comment (ajkl2533) [inline visual-regressions/Dockerfile:1]: Shouldn't we add it to SSC docker registry?

Comment (abykovssc) [reply visual-regressions/Dockerfile:1]: we have an image in a private registry, just not in public one, and I'm not sure we even have a public registry 🤔

**Review by ajkl2533 — APPROVED** [review]

**Pull request merged**